pkg/asset/manifests/azure/cloudproviderconfig.go

```diff
@@ -24,19 +26,32 @@ type CloudProviderConfig struct {
 // JSON generates the cloud provider json config for the azure platform.
 // managed resource names are matching the convention defined by capz
 func (params CloudProviderConfig) JSON() (string, error) {
+	azureStack := params.CloudName.Name() == string(azure.StackCloud)
```

Comment:
Does this need to be this complicated?
```suggestion
	azureStack := params.CloudName == azure.StackCloud
```